app/src/parser.c

```diff
@@ -113,24 +119,69 @@ parser_error_t parser_getItem(const parser_context_t *ctx,
     CHECK_ERROR(checkSanity(numItems, displayIdx))
     cleanOutput(outKey, outKeyLen, outVal, outValLen);
 
-    switch (displayIdx)
+    uint8_t txItems = 0;
+    CHECK_ERROR(parser_getTxNumItems(ctx, &txItems))
+
+    const uint8_t common_items = _getNumCommonItems();
+
+    if (displayIdx < common_items) {
+        return print_common_items(ctx, displayIdx, outKey, outKeyLen,
+                                  outVal, outValLen, pageIdx, pageCount);
+    }
+
+    displayIdx-=common_items;
+    uint8_t txDisplayIdx = 0;
+    
+    switch (ctx->tx_obj->module_id)
     {
-        case 0: {
-            uint8_t hash[32] = {0};
-            char hash_str[100] = {0};
-            CHECK_ERROR(crypto_hash(ctx->buffer, ctx->bufferLen,
-                                    hash, sizeof(hash)))
-            array_to_hexstr((char*) &hash_str, sizeof(hash_str),
-                            (const uint8_t*) &hash, sizeof(hash));
-
-            // Display Item 0
-            snprintf(outKey, outKeyLen, "Txn hash");
-            pageString(outVal, outValLen, (const char*) &hash_str, pageIdx, pageCount);
-            return parser_ok;
-        }
-
-    default:
-        break;
+        case TX_MODULE_ID_TOKEN:
+            switch (ctx->tx_obj->command_id)
+            {
+                case TX_COMMAND_ID_TRANSFER:
+                    return print_module_token_transfer(ctx, displayIdx, outKey, outKeyLen,
+                                      outVal, outValLen, pageIdx, pageCount);
+                case TX_COMMAND_ID_CROSSCHAIN_TRANSFER:
+                    return print_module_token_cross(ctx, displayIdx, outKey, outKeyLen,
+                                      outVal, outValLen, pageIdx, pageCount);
+                default:
+                    return parser_unexpected_value; 
+            }
+        case TX_MODULE_ID_AUTH:
+            CHECK_ERROR(getItem(displayIdx, &txDisplayIdx))
+            return print_module_auth_reg(ctx, displayIdx, txDisplayIdx, outKey, outKeyLen,
+                                      outVal, outValLen, pageIdx, pageCount);
+            break;
+
+        case TX_MODULE_ID_DPOS:
+            switch (ctx->tx_obj->command_id)
+            {
+                case TX_COMMAND_ID_REGISTER_DELEGATE:
+                    return print_module_dpos_reg_delegate(ctx, displayIdx, outKey, outKeyLen,
+                                      outVal, outValLen, pageIdx, pageCount);
+
+                case TX_COMMAND_ID_VOTE_DELEGATE:
+                    CHECK_ERROR(getItem(displayIdx, &txDisplayIdx))
+                    return print_module_dpos_vote(ctx, displayIdx,  txDisplayIdx, outKey, outKeyLen,
+                                       outVal, outValLen, pageIdx, pageCount);
+
+                case TX_COMMAND_ID_UNLOCK_TOKEN:
+                    return parser_ok;
```

Comment:
We should print something here. Otherwise the display will show **???** If there is nothing to display, then we are counting an extra element.
The same applies to: **TX_COMMAND_ID_REPORT_DELEGATE_MISBEHAVIOUR**